Med 11

Crasheroo opened the pull request.

Crasheroo committed:
+ map do nowej linijki

Crasheroo committed:
+ password change functionality

Crasheroo committed:
+ password change functionality

Crasheroo committed:
+ cleanup

Crasheroo committed:
+ cleanup

Crasheroo committed:
+ cleanup

Crasheroo committed:
+ cleanup

Crasheroo committed:
+ cleanup

Crasheroo committed:
Merge branch 'feature/MED-04' into feature/MED-11

Crasheroo committed:
+ cleanup

Crasheroo committed:
+ cleanup

konradgab commented on src/main/java/com/example/medicalclinic/model/PasswordRequest.java:
```diff
@@ -0,0 +1,8 @@
+import lombok.Data;
+
+@Data
+public class PasswordRequest {
```
czemu nie rekord?

konradgab commented on src/main/java/com/example/medicalclinic/repository/PatientRepository.java:
```diff
@@ -27,20 +26,40 @@ public Optional<Patient> findByEmail(String email) {
-        });
+    public Patient updatePasswordByEmail(String email, String password) {
+        return findByEmail(email)
+                .map(patient -> {
```
czemu map?

konradgab commented on src/main/java/com/example/medicalclinic/repository/PatientRepository.java:
```diff
@@ -27,20 +26,40 @@ public Optional<Patient> findByEmail(String email) {
+                    patient.setPassword(password);
+                    return patient;
+                })
+                .orElseThrow(() -> new PatientException("Patient with email: " + email + " not found"));
```
moim zdaniem tutaj nie powinnismy rzucic takiego wyjątku

Crasheroo committed:
+ not allowing to add patient with already used idCardNo
+ not allowing to add patient with already used email
+ not allowing to edit patient with already used email
+ not allowing to edit patient with already used idCardNo
+ moved logic to Respoitory from Service

Crasheroo committed:
+ PatientDTO

Crasheroo committed:
+ PatientDTO

Crasheroo committed:
+ MapStruct
+ Returning message when deleting patient
+ not allowing to return password of patient/s to user

Crasheroo committed:
+ Returning patient's fullName to user
+ mapstruct fix in pom

Crasheroo committed:
+ Returning patient's fullName to user
+ mapstruct fix in pom

konradgab commented on pom.xml:
```diff
@@ -35,6 +35,24 @@
+		<dependency>
+			<groupId>org.mapstruct</groupId>
+			<artifactId>mapstruct</artifactId>
+			<version>1.6.3</version>
```
uzyj zmiennej w pomie

konradgab commented on src/main/java/com/example/medicalclinic/controller/PatientController.java:
```diff
@@ -13,31 +18,39 @@
 
     @GetMapping
-    public List<Patient> getPatients() {
+    public List<PatientDTO> getPatientsDTO() {
```
gePatients

konradgab commented on src/main/java/com/example/medicalclinic/controller/PatientController.java:
```diff
@@ -13,31 +18,39 @@
-        patientService.removePatientByEmail(email);
+    public ResponseEntity<ResponseMessage> removePatient(@PathVariable("email") String email) {
+        ResponseMessage responseMessage = patientService.removePatientByEmail(email);
+        return ResponseEntity.ok(responseMessage);
```
uzywasz rest controllera po co zwracasz ResponseEntityty dodatkowo tutaj zauwaz ze zwracasz status Ok  a przy pomocy adnotacji mowisz ze ma byc NO_CONTENT popraw

konradgab commented on src/main/java/com/example/medicalclinic/mapper/PatientMapper.java:
```diff
@@ -0,0 +1,28 @@
+
+import java.util.List;
+
+@Mapper
```
@Mapper(componentModel = "spring") <- dzieki temu nasz wygenerowany mapper (jego implementacja) bedzie mial nad soba adntoacje @Component dzieki czemu bedzie beanem i bedziemy mogli go wstrzykiwac

konradgab commented on src/main/java/com/example/medicalclinic/mapper/PatientMapper.java:
```diff
@@ -0,0 +1,28 @@
+
+@Mapper
+public interface PatientMapper {
+    PatientMapper INSTANCE = Mappers.getMapper(PatientMapper.class);
```
Uzywaj i wstrzykuj beana do mappera a nie w taki sposob

konradgab commented on src/main/java/com/example/medicalclinic/repository/PatientRepository.java:
```diff
@@ -17,30 +18,93 @@ public List<Patient> getPatients() {
         return List.copyOf(patients);
     }
 
+    public List<PatientDTO> getPatientsAsDTO() {
```
repozytorium nie powinno zwracac DTO

konradgab commented on src/main/java/com/example/medicalclinic/repository/PatientRepository.java:
```diff
@@ -17,30 +18,93 @@ public List<Patient> getPatients() {
+                .findFirst();
+    }
+
+    public PatientDTO getPatientDTOByEmail(String email) {
```
repozytorium nie powinno zwracac DTO

Crasheroo committed:
+ fixes

Crasheroo committed:
+ fixes

Crasheroo committed:
+ fixes
